### PR TITLE
Tests: faillock command does not read faillock.conf option

### PR DESCRIPTION
**Pull request by aborah-sudo**
faillock command does not read faillock.conf option

### CODE REVIEW

File: multihost_test/bz_automation/test_automation.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from sssd.testlib.common.utils import SSHClient
+
+
+def execute_cmd(multihost, command):
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+@pytest.mark.tier1
+class TestShadowBz(object):
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:16]: I guess this should be `TestPamBz`

File: multihost_test/bz_automation/test_automation.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+                      password="bad_pass")
+        #assert 'V' in execute_cmd(multihost, "faillock --user local_anuj").stdout_text
+        #execute_cmd(multihost, "faillock --user local_anuj --reset")
+        #assert 'V' not in execute_cmd(multihost, "faillock --user local_anuj").stdout_text
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(multihost.client[0].ip, username="local_anuj",
+                      password="bad_pass")
+        assert 'V' in execute_cmd(multihost, "faillock --dir "
+                                             "/var/log/faillock "
+                                             "--user local_anuj").stdout_text
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:47]: The tally directory shouldn't be specified in the command line as faillock should read it from the configuration file that you modified in the previous lines.

File: multihost_test/bz_automation/test_automation.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+        execute_cmd(multihost, "faillock --dir "
+                               "/var/log/faillock "
+                               "--user local_anuj --reset")
+        assert 'V' in execute_cmd(multihost, "faillock --dir "
+                                             "/var/log/faillock "
+                                             "--user local_anuj").stdout_text
+        execute_cmd(multihost, "faillock --dir "
+                               "/var/log/faillock --user "
+                               "local_anuj --reset")
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:56]: I don't understand the extra calls to `faillock`. Why are you doing them?

**Review by ikerexxe — CHANGES_REQUESTED** [review]

File: multihost_test/bz_automation/test_automation.py (hunk 2 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:47]: Here i am adding extra line of tests , just to make sure all stuff works

File: multihost_test/bz_automation/test_automation.py [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+                               "/etc/security/faillock.conf")
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(multihost.client[0].ip, username="local_anuj",
+                      password="bad_pass")
+        #assert 'V' in execute_cmd(multihost, "faillock --user local_anuj").stdout_text
+        #execute_cmd(multihost, "faillock --user local_anuj --reset")
+        #assert 'V' not in execute_cmd(multihost, "faillock --user local_anuj").stdout_text
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(multihost.client[0].ip, username="local_anuj",
+                      password="bad_pass")
```
[the remark this answers is not in the document]
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:44]: here i am making sure , logs are there with multiple failure

File: multihost_test/bz_automation/test_automation.py (hunk 1 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:16]: changed

File: multihost_test/bz_automation/test_automation.py (hunk 3 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:56]: Here i am adding extra line of tests , just to make sure all stuff works

File: multihost_test/bz_automation/test_automation.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (ikerexxe) [reply multihost_test/bz_automation/test_automation.py:44]: I see what you are doing but it's difficult to understand. There are like 3 different test cases in this function. Do you mind adding comments before each section to separate them and explain what you are doing?

File: multihost_test/bz_automation/test_automation.py (hunk 2 above)
Comment (ikerexxe) [reply multihost_test/bz_automation/test_automation.py:47]: See previous comment.

File: multihost_test/bz_automation/test_automation.py (hunk 3 above)
Comment (ikerexxe) [reply multihost_test/bz_automation/test_automation.py:56]: See previous comment.

File: multihost_test/bz_automation/test_automation.py (hunk 2 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:53]: Done

File: multihost_test/bz_automation/test_automation.py (hunk 4 above)
[the remark this answers is not in the document]
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:49]: Done

File: multihost_test/bz_automation/test_automation.py (hunk 3 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py]: Done

File: multihost_test/bz_automation/test_automation.py [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+        assert 'Valid' in execute_cmd(multihost, "faillock --dir /var/log/faillock "
+                                                 "--user local_anuj").stdout_text.split('\n')[1]
+        assert 'V' in execute_cmd(multihost, "faillock --dir "
+                                             "/var/log/faillock "
+                                             "--user local_anuj").stdout_text.split('\n')[2]
+        # check --reset command with directory along with other prefixes
+        execute_cmd(multihost, "faillock --dir "
+                               "/var/log/faillock "
+                               "--user local_anuj --reset")
+        # check if command workes again after reset
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:62]: s/workes/works

File: multihost_test/bz_automation/test_automation.py [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+        assert 'Valid' in execute_cmd(multihost, "faillock --dir /var/log/faillock "
+                                                     "--user local_anuj").stdout_text.split('\n')[1]
+        assert 'V' not in execute_cmd(multihost, "faillock --dir "
+                                             "/var/log/faillock "
+                                             "--user local_anuj").stdout_text.split('\n')[2]
+        # Just claning and backup
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:68]: s/claning/cleaning
s/backup/restoring

File: multihost_test/bz_automation/test_automation.py [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+        with pytest.raises(paramiko.ssh_exception.AuthenticationException):
+            SSHClient(multihost.client[0].ip, username="local_anuj",
+                      password="bad_pass")
+        # Tests faillock with prefix --user
+        assert 'Valid' in execute_cmd(multihost, "faillock --user local_anuj").stdout_text.split('\n')[1]
+        assert 'V' in execute_cmd(multihost, "faillock --user local_anuj").stdout_text.split('\n')[2]
+        # check --reset command without directory
+        execute_cmd(multihost, "faillock --user local_anuj --reset")
+        # check above command worked
+        assert 'Valid' in execute_cmd(multihost, "faillock --user local_anuj").stdout_text.split('\n')[1]
```
Comment (ikerexxe) [inline multihost_test/bz_automation/test_automation.py:46]: If you clean the tally file, then there shouldn't be any content. So, why are you checking for the `Valid` string in that content? Are you sure that this should work?

**Review by ikerexxe — COMMENTED** [review]
I've added some minor comments.

Thanks for clarifying my previous concerns with the comments.

File: multihost_test/bz_automation/test_automation.py (hunk 7 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:46]: Removed the Valid part but after cleaning V should not be there , so i will keep that as it is

File: multihost_test/bz_automation/test_automation.py (hunk 5 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:62]: fixed

File: multihost_test/bz_automation/test_automation.py (hunk 6 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py:68]: fixed

**Comment by aborah-sudo** [comment]
CI passed :  https://ci-jenkins-csb-idmops.apps.ocp-c1.prod.psi.redhat.com/job/trigger-test-suite-tool/5429/console

File: multihost_test/bz_automation/test_automation.py (hunk 7 above)
Comment (aborah-sudo) [reply multihost_test/bz_automation/test_automation.py]: CI passed :  https://ci-jenkins-csb-idmops.apps.ocp-c1.prod.psi.redhat.com/job/trigger-test-suite-tool/5429/console

Comment (ikerexxe) [reply multihost_test/bz_automation/test_automation.py]: yes, that makes sense now

**Review by ikerexxe — APPROVED** [review]

**Pull request merged**